hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/accessor/AccountDatabaseAccessor.java

```diff
@@ -213,9 +215,18 @@ private Supplier<TokenAccountBalances> getNumberOfAllAndPositiveBalanceTokenAsso
         return Suppliers.memoize(() -> new TokenAccountBalances(allAggregated, positiveAggregated));
     }
 
-    private JKey parseJkey(byte[] keyBytes) {
+    private JKey parseJkey(byte[] keyBytes, Long entityId, boolean isContract) {
         try {
-            return keyBytes == null ? null : asFcKeyUnchecked(Key.parseFrom(keyBytes));
+            if (keyBytes == null) {
+                if (isContract) {
+                    return new JContractIDKey(
+                            ContractID.newBuilder().setContractNum(entityId).build());
+                }
+                return null;
+            } else {
+                return asFcKeyUnchecked(Key.parseFrom(keyBytes));
```

Comment:
It's incorrect to map `entity.getId()` to just the number of `ContractID` since id contains shard and realm encoded in it as well. Instead, pass entire `Entity` and create `ContractID` from separate realm, shard and num fields.

```suggestion
            var keyBytes = entity.getKey();
            if (keyBytes == null) {
                if (entity.getType() == EntityType.CONTRACT) {
                    return new JContractIDKey(                            ContractID.newBuilder()
                        .setContractNum(entity.getNum())
                        .setRealmNum(entity.getRealm())
                        .setShardNum(entity.getShard())
                        .build());
                }
                return null;
            } else {
                return asFcKeyUnchecked(Key.parseFrom(keyBytes));
```